src/modules/user/services/userEnergy/user.energy.compute.service.ts

```diff
@@ -167,6 +168,14 @@ export class UserEnergyComputeService {
         userAddress: string,
         contractAddress: string,
     ): Promise<OutdatedContract> {
+        const isProducingRewards = await this.stakingCompute.isProducingRewards(
+            contractAddress,
+        );
+
+        if (!isProducingRewards) {
```

Comment:
Put this check down below and add ` && !outdatedClaimProgress`. This check is used to generate a claim transaction so in case the staking isn't generating rewards but the user still has to claim some from previous weeks, he would be unable to do that

Reply:
done